src/points/PointsHistoryBottomSheet.test.tsx

```diff
@@ -0,0 +1,122 @@
+    default: react.View,
+    BottomSheetSectionList: react.SectionList,
+  }
+})
+
+const MOCK_RESPONSE_NO_NEXT_PAGE: GetHistoryResponse = {
+  data: [
+    {
+      activity: 'swap',
+      points: '20',
+      createdAt: '2024-03-05T19:26:25.000Z',
+      metadata: {
+        to: 'celo-alfajores:native',
+        from: 'celo-alfajores:0x874069fa1eb16d44d622f2e0ca25eea172369bc1',
+      },
+    },
+    {
+      activity: 'swap',
+      points: '20',
+      createdAt: '2024-01-04T19:26:25.000Z',
+      metadata: {
+        to: 'celo-alfajores:0xe4d517785d091d3c54818832db6094bcc2744545',
+        from: 'celo-alfajores:native',
+      },
+    },
+  ],
+  hasNextPage: false,
+  nextPageUrl: '',
+}
+
+describe(PointsHistoryBottomSheet, () => {
+  const mockFetch = fetch as FetchMock
+  beforeEach(() => {
+    jest.useRealTimers()
+    jest.clearAllMocks()
+    mockFetch.resetMocks()
+  })
+
+  function renderScreen(storeOverrides: RecursivePartial<RootState> = {}) {
+    const store = createMockStore({
+      ...storeOverrides,
+    })
+
+    const dispatch = jest.spyOn(store, 'dispatch')
+
+    const tree = render(
+      <Provider store={store}>
+        <PointsHistoryBottomSheet forwardedRef={{ current: null }} />
+      </Provider>
+    )
+
+    return {
+      dispatch,
+      store,
+      ...tree,
+    }
+  }
+
+  it('Displays content when idle', async () => {
+    const tree = renderScreen()
+
+    expect(tree.getByTestId('PointsHistoryBottomSheet/MainContent')).toBeTruthy()
+    expect(tree.queryByTestId('PointsHistoryBottomSheet/Loading')).toBeNull()
+  })
+
+  it('Displays content while loading', async () => {
+    const tree = renderScreen({
+      points: { pointsHistory: MOCK_RESPONSE_NO_NEXT_PAGE.data, getHistoryStatus: 'loading' },
+    })
+    await waitFor(() => expect(tree.getByTestId('PointsHistoryList').props.data.length).toBe(2))
+
+    expect(tree.getByTestId('PointsHistoryBottomSheet/MainContent')).toBeTruthy()
+    expect(tree.getByTestId('PointsHistoryBottomSheet/Loading')).toBeTruthy()
+  })
+
+  it('fetches more content when bottom is visible', async () => {
+    const { dispatch, getByTestId } = renderScreen({
+      points: {
+        pointsHistory: MOCK_RESPONSE_NO_NEXT_PAGE.data,
+        getHistoryStatus: 'idle',
+        nextPageUrl: 'foo',
+      },
+    })
+
+    // Trigger end reached to fetch more
+    getByTestId('PointsHistoryList').props.onEndReached()
+
+    expect(dispatch).toHaveBeenCalledWith(getHistoryStarted({ getNextPage: true }))
+  })
+
+  it('Shows error screen if fetch fails', async () => {
+    const tree = renderScreen({ points: { getHistoryStatus: 'error' } })
+    expect(tree.getByTestId('PointsHistoryBottomSheet/ErrorState')).toBeTruthy()
+  })
+
+  it('Dispatches action when try again is pressed', async () => {
+    const { dispatch, getByTestId } = renderScreen({ points: { getHistoryStatus: 'error' } })
+    fireEvent.press(getByTestId('PointsHistoryBottomSheet/TryAgain'))
```

Comment:
nit: if we query by text, we ensure that the user will see the correct labels (Kathy already mentioned it in another PR comment)
i personally try to resort to test ids only if it's impossible to query by text.
see also: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-querying-by-text